### PR TITLE
Places search added and place pages

### LINKED ISSUE #191: Add search functionality to website
We should add a search function so people can find a place without using their device's location. The back-end logic of this will start on our server, and be added to the website first. The logic on our server can then be reused for the same feature on our app.

### PR DESCRIPTION
**Pull request by ajones55555**
closes https://github.com/Peer-Stevens/peer/issues/191

### CODE REVIEW

**Review by elenirotsides — COMMENTED** [review]
A couple of questions:

- Is it possible to move the `x` further to the right of the input field? 
image
- Do you intend on only showing one place in the search results? The place that it also shows me is typically no where near my location and not even in the same state; is it possible to implement the `zip code` and/or `city` input field(s) like we talked about the other day so we can narrow the search?
- Can you make the result _feel_ clickable to the user? The only reason why I knew to click on the result was because you mentioned it in this PR. Otherwise, its not really clear that there is further information to be presented past this page.

LMK what you think!

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+NEXT_PUBLIC_BACKEND_URL=https://peer-server-stevens.herokuapp.com
```
Comment (ajones55555) [inline .env]: didn't mean to commit this so i'm removing it, but luckily this value isn't sensitive, so we don't need to worry about removing the historical file

**Comment by elenirotsides** [comment]
I love the changes! I'm still wondering why we're only showing one result though? Is there a reason for this? @aej11a

**Review by elenirotsides — COMMENTED** [review]
Can you please restrict the `zip code` input field to just numbers? What I did below totally works, but its not what we intend it to do:

image

I think a 'nice to have' would also be including a city field. I personally cannot remember zip codes and I'm sure there are other people out there like myself, so having both a `zip code` and `city` field is ideal so that users can fill out what they want.

**Comment by ajones55555** [comment]
> I love the changes! I'm still wondering why we're only showing one result though? Is there a reason for this? @aej11a

I'm actually not 100% why but the API only returns 1 or two results. Maybe we can configure it to be less precise? Do you think this is a major issue?

**Comment by elenirotsides** [comment]
I don't think its an issue for right now, but we should definitely create an issue for this so we can revisit it

**Comment by ajones55555** [comment]
>  I personally cannot remember zip codes and I'm sure there are other people out there like myself

Remember though that this is 99% for business-owners or managers. I would find it pretty odd if people didn't know the zip code of their own building, especially because of all the paperwork related to running a business.

**Comment by ajones55555** [comment]
Maybe as a sort-of-solution for both of those issues, we could rename the name field to just "search" and include some hint like "enter the place name and address"
What do you think?

**Comment by elenirotsides** [comment]
> Remember though that this is 99% for business-owners or managers. I would find it pretty odd if people didn't know the zip code of their own building, especially because of all the paperwork related to running a business.

This is a good point! Regardless, I still think it would be nice to have a city field. Just my opinion though

**Comment by ajones55555** [comment]
or maybe we should even do a full-blown address form? idk

**Comment by elenirotsides** [comment]
> Maybe as a sort-of-solution for both of those issues, we could rename the name field to just "search" and include some hint like "enter the place name and address"

So if we do something like this (which I think is the ideal solution btw), I think we need to make sure that the results aren't being restricted to just one, since cities can have many locations for a chain establishment (think McDonalds in NYC for example). Perhaps you can change the field to look something like this (see pic below) and then we can worry about the API's response in a later issue. Or, we can do a full blown address form.

image

**Comment by ajones55555** [comment]
ok, maybe i'll try out the address form tomorrow evening

**Review by elenirotsides — APPROVED** [review]

**Pull request merged**